From 9e7510ddfa6954a7480a6d6153b3a1a3aea75ce7 Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Tue, 21 Apr 2020 23:14:14 +0300
Subject: [PATCH 1/2] [gecko] Configuration option. JB#49613

All js and core configuration options should be
defined in this patch.

- With 32bit arm (Linaro GCC 8.2-2018.08) startup of of the context
failed similarly as https://bugzilla.mozilla.org/show_bug.cgi?id=1337263
Also MacOS builds were reporting similar failures but those were fixed
differently.

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 js/src/old-configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/src/old-configure.in b/js/src/old-configure.in
index 1c5c9e2145f6..3105458994a9 100644
--- a/js/src/old-configure.in
+++ b/js/src/old-configure.in
@@ -660,7 +660,7 @@ case "$target" in
 *-*linux*)
     if test "$GNU_CC" -o "$GNU_CXX"; then
         MOZ_PGO_OPTIMIZE_FLAGS="-O3"
-        MOZ_OPTIMIZE_FLAGS="-O3"
+        MOZ_OPTIMIZE_FLAGS="-O3 -fno-schedule-insns"
         if test -z "$CLANG_CC"; then
            MOZ_OPTIMIZE_FLAGS="-freorder-blocks $MOZ_OPTIMIZE_FLAGS"
         fi
-- 
2.25.2

